### DISCUSSION TITLE
Why 11ty does not display anything?

### DISCUSSION THREAD

**Discussion by cawa-93** [11ty/eleventy#2361] [Q&A] [answered]
I try show all posts on home page.
[`/index.njk`:](https://github.com/cawa-93/it-ua-hub/blob/a3615493e0a2a6817cbe8afdebf50ce8d1fea3e6/index.njk)
```njk
---
layout: layouts/post-feed.njk
postslist: {{ collections.posts }}
---
```

[`/layouts/post-feed.njk`:](https://github.com/cawa-93/it-ua-hub/blob/a3615493e0a2a6817cbe8afdebf50ce8d1fea3e6/_includes/layouts/post-feed.njk)
```njk
{{ postslist.length }}
{% for post in postslist %}
  <article>
    <h2><a href="{{ post.url | url }}">{{ post.data.title }}</a></h2>
  </article>
{% endfor %}
```

And I get a white screen. [Posts in the project are present](https://github.com/cawa-93/it-ua-hub/tree/a3615493e0a2a6817cbe8afdebf50ce8d1fea3e6/posts), but the eleventh just quietly does not display anything. And how to fix it? There are no errors in the console.

**Comment by cfjedimaster** [comment, marked as answer]
You can't assign to front matter like that - from the docs here, https://www.11ty.dev/docs/data-frontmatter/, it says:

"Note that only the [permalink](https://www.11ty.dev/docs/permalinks/) and [eleventyComputed](https://www.11ty.dev/docs/data-computed) front matter values can contain variables and shortcodes like you would use in the body of your templates. If you need to use variables or shortcodes in other front matter values, use eleventyComputed to set them."

So you could use eleventyComputed I believe, or just use collections.posts in your for loop (it would actually be less code then :)

**Reply by cawa-93** [reply]
Sounds reasonable. But it still doesn't work. I tried:
`index.njk`
```njk
---
layout: layouts/post-feed.njk
eleventyComputed:
  postslist: {{ collections.posts }}
---
```

**Reply by cfjedimaster** [reply]
I believe it has to be in quotes, specifically {{ ... }}. Sorry to not test this myself - it's kid drop off time. :

**Reply by cawa-93** [reply]
> I believe it has to be in quotes, specifically {{ ... }}. Sorry to not test this myself - it's kid drop off time. :

And now `postlist` is a string `"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"`. And for loop iterate it by letters

**Reply by cawa-93** [reply]
I have no idea how to adapt it in yaml format, but I managed to run it only in JS syntax
```js
---js
{
  "layout": "layouts/post-feed.njk",
  "eleventyComputed": {
    "postslist": (data) => data.collections.posts
  }
}
---
```

**Reply by cfjedimaster** [reply]
Cool - glad you got it. I honestly thought it could work in YAML - going to investigate more when I have time.

**Reply by cfjedimaster** [reply]
I've been able to test and can confirm - for both collection and global data, eleventyComputed doesn't let you make an alias. Still digging.

**Reply by cfjedimaster** [reply]
Ok... I stand corrected. I had tabs in my front matter by accident. Ok so get this. If you have a global data file as an array, and use it in eleventyComputed, you get the toString() version of it, not the actual array. So global data... kinda works? Going to file a bug report on this.

**Reply by cfjedimaster** [reply]
I filed https://github.com/11ty/eleventy/issues/2363. Note that if your intent is to make an alias of the collection for simpler template code, you can also do something like this: `{% assign myposts = collections.post %}`
